content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+
+h4(#rest-or-realtime). Remain connected, or not
+
+With the availability of both the "Realtime":/realtime and "REST":/rest libraries, choose what suits your application's needs best and get started.
+
+h4(#auth). Choose the right auth scheme
+
+Use "Basic Authentication":/general/authentication#basic-authentication or "Token Authentication":/general/authentication#token-authentication as well as ensure security by "encrypting your messages.":/realtime/encryption
+
+h3(#using-ably-like-a-pro). Using Ably like a pro
+
+h4. 1. Attaching vs subscribing to a channel
+
+* When your client merely attaches to a channel on Ably, the platform will start sending message data even if your client has not subscribed to the channel. Subscribing to a channel simply registers a listener using which your client can perform additonal operations, however, Ably has not knowledge about this as it only listens for a client to attach to a channel. Bear this in mind while attaching to a channel for publish-only purpose.
+
+h4. 2. Using separate channels when you want messages to be sent to different sets of subscribers.
+
+* Ably pushes all the data published on a channel to every subscriber attached to that channel. Subscribers can choose to only subscribe to events with a certain message name, but that filtering is done on the client side (however, it's useful for things like letting the subscriber trigger different listeners for different events). If you have messages that don't need to be sent to some clients at all, you're often better off using a separate channel for those instead of implementing a filter. This way, you won't be sending unnecessary messages that count towards your package limit.
+
+h4. 3. Being present without subscribing to presence events
+
+* If you want lots of people to be present, but not necessarily listen for "presence change events":/realtime/presence#presence-states, you should use token capabilities to prevent all users from receiving presence notifications while allowing them to enter only. This is a common scenario when you just want to know how many people are present on a channel at any point in time.
+
+h4. 4. Handling connection and channel state&#45;change events
+
+* Ably's Realtime SDKs automatically handle disruptions such as network disconnections, so you don't need to do anything. The library emits "connection":/realtime/connection#connection-states and "channel":/realtime/channels-messages#channel-stateschannel state change events that you can listen to in case you'd like be informed about what it's doing (for example, if you want to know whenever the library has lost message continuity on a channel due to being disconnected for more than two minutes). But there's no need to take any particular action in response to any of these, the library will automatically reconnect and reattach to channels after the network disruption is over.
```

Comment:
Well they do need to do things, just not in regards to manually reconnecting.  Be explicit, not vague please

Reply:
I don't think you should bundle connection and channel events into one section.  They have different states, different events, and we should aim to explain this.  Perhaps provide a simple example in code for each (Javasceript is fine).  
`/realtime/channels-messages#channel-stateschannel` is not a validf link.

"But there's no need to take any particular action in response to any of these, the library will automatically reconnect and reattach to channels after the network disruption is over."  Well there is if it fails.  This is really not achieving the goal of this document I am afraid. We should be advising them that handling state change events may be useful to them because they need to reflect this state to their users, and also they need to handle failures such as connection failures (because of auth problems etc) or channel failures (permission issues etc).